piccolo/columns/reverse_lookup.py

```diff
@@ -0,0 +1,226 @@
+        table: t.Type[Table],
+    ):
+        """
+        :param columns:
+            Which columns to include from the related table.
+        :param as_list:
+            If a single column is provided, and ``as_list`` is ``True`` a
+            flattened list will be returned, rather than a list of objects.
+        :param load_json:
+            If ``True``, any JSON strings are loaded as Python objects.
+        :param table:
+            Parent table for reverse lookup.
+
+        """
+        self.as_list = as_list
+        self.columns = columns
+        self.reverse_lookup = reverse_lookup
+        self.load_json = load_json
+        self.table = table
+
+        safe_types = [int, str]
+
+        # If the columns can be serialised / deserialise as JSON, then we
+        # can fetch the data all in one go.
+        self.serialisation_safe = all(
+            (column.__class__.value_type in safe_types)
+            and (type(column) not in (JSON, JSONB))
+            for column in columns
+        )
+
+    @property
+    def foreign_key_columns_index(self) -> int:
+        reverse_lookup_table = (
+            self.reverse_lookup._meta.resolved_reverse_joining_table
+        )
+        fk_columns = [
+            i._meta.name
+            for i in reverse_lookup_table._meta.foreign_key_columns
+        ]
+        return fk_columns.index(self.table._meta.tablename)
+
+    def get_select_string(self, engine_type: str, with_alias=True) -> str:
+        reverse_lookup_table = (
+            self.reverse_lookup._meta.resolved_reverse_joining_table
+        )
+        reverse_lookup_table_name = reverse_lookup_table._meta.tablename
+        reverse_lookup_pk = reverse_lookup_table._meta.primary_key._meta.name
+        reverse_lookup_fk_table_name = (
+            reverse_lookup_table._meta.foreign_key_columns[
+                self.foreign_key_columns_index
+            ]._meta.name
+        )
+        reverse_lookup_fk = reverse_lookup_table._meta.foreign_key_columns[
+            self.foreign_key_columns_index
+        ]._meta.db_column_name
+
+        reverse_select = f"""
+            "{reverse_lookup_table_name}"
+            JOIN "{reverse_lookup_fk_table_name}" "inner_{reverse_lookup_fk_table_name}" ON (
+                "{reverse_lookup_table_name}"."{reverse_lookup_fk}"
+                = "inner_{reverse_lookup_fk_table_name}"."{reverse_lookup_pk}"
+            ) WHERE "{reverse_lookup_table_name}"."{reverse_lookup_fk}"
+                = "{reverse_lookup_fk_table_name}"."{reverse_lookup_pk}"
+        """  # noqa: E501
+
+        if engine_type == "postgres":
+            if self.as_list:
+                column_name = self.columns[0]._meta.db_column_name
+                return f"""
+                    ARRAY(
+                        SELECT
+                            "{reverse_lookup_table_name}"."{column_name}"
+                        FROM {reverse_select}
+                    ) AS "{reverse_lookup_table_name}s"
```

Comment:
I think we should use `AS "{original_column_name}s"`. The table must give us that name. See m2m.py: https://github.com/piccolo-orm/piccolo/blob/14012e59759c849ad732b4c4e4fb40636dc88aa6/piccolo/columns/m2m.py#L93